inspirehep/modules/theme/views.py

```diff
@@ -549,3 +550,18 @@ def menu_fixup():
         current_menu.submenu("settings.oauthclient").hide()
 
     current_app.before_first_request_funcs.append(menu_fixup)
+
+
+#
+# Redirect /record/N
+#
+
+@blueprint.route('/record/<control_number>')
+def record(control_number):
+    try:
+        pid = PersistentIdentifier.query.filter_by(
+            pid_value=control_number).one()
+        return redirect('/{collection}/{control_number}'.format(
+            collection=pid.pid_type, control_number=control_number)), 301
+    except Exception:
```

Comment:
Since we are falling back to SQLAlchemy I guess we can catch `SQLAlchemyError`. On the other hand I wasn't entirely sure if the second line could fail in weirder ways...